prometrics.go

```diff
@@ -0,0 +1,75 @@
+package prometrics
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type Registry struct {
+	reg prometheus.Registerer
+	gat prometheus.Gatherer
+}
+
+func NewRegistry() *Registry {
+	r := prometheus.NewRegistry()
+	r.MustRegister(prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}))
+	r.MustRegister(prometheus.NewGoCollector())
+	return &Registry{
+		reg: r,
+		gat: r,
+	}
+}
+
+func (r *Registry) ReportDuration() func(*gin.Context) {
+	duration := prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "request_duration_seconds",
+			Help:    "Display duration by code, method, and route",
+			Buckets: prometheus.DefBuckets,
```

Comment:
Hardcoding buckets inherently makes this inflexible. Allow clients to specify their own buckets instead